### DISCUSSION TITLE
change the wallpaper choose app preview style

### DISCUSSION THREAD

**Discussion by gitttyyy** [end-4/dots-hyprland#338] [Q&A] [answered]
I find @midn8hustlr awesome work with #337,and then  I find the  better image preview with nsxid. So I want this.
![image](https://github.com/end-4/dots-hyprland/assets/154318356/3f009b8d-1509-4823-b8dd-c980c948b95d)
And now mine is like this
![image](https://github.com/end-4/dots-hyprland/assets/154318356/56388af9-1036-498c-bcac-bb00fdfa41c2)
Is it change the swichwall.sh? I`m not sure how to change it correctly with nsxid or yad.I find may config nsxid is needed.
Can you share you config? @midn8hustlr 😍

**Comment by midn8hustlr** [comment, marked as answer]
Yes you are absolutely right. You just need to replace the following line in switchwall.sh
```bash
imgpath=$(yad --width 1200 --height 800 --file --title='Choose wallpaper')
```
with
```bash
imgpath=$(nsxiv -r ~/walls -t --alpha-layer -o -b)
```
Replace the ~/walls directory with the directory where your put you wallpapers in.
On hitting the switchwall keybind, just select the wallpaper and apply using Shift+Q.
Make sure you have nsxiv installed.
```bash
sudo pacman -Syu nsxiv
```

The background of nsxiv is white by default. To change it to dark (or pitch black like me), create a file ~/.Xresources and put the following lines in it.
```
Xcursor.theme: Bibata-Modern-Classic
*background: #000000
*foreground: #ffffff
```
Execute on startup or put the following in hypr/hyprland/execs.conf.
```bash
xrdb ~/.Xresources
```

**Reply by gitttyyy** [reply]
Thank you😍

**Comment by midn8hustlr** [comment]
@end-4 In fact, what do you think about using an image viewer instead of a file picker for wallpapers?
Ik, its not portable enough, you can only apply wallpapers stored in some specific directory.
I am also trying to find a wayland equivalent of nsxiv, do you have any minimal image viewer in mind?

**Reply by end-4** [reply]
i like how you have a grid layout with previews. but not being able to select pictures outside of a specific location isn't that great. 
so i wanna find something else.
or figure out how to use grid layout in yad...

**Reply by midn8hustlr** [reply]
Actually 'nsxiv -r' is recursive
So you can select multiple directories to search from and it will search all of them till full depth
In my walls directory, I have all my wallpapers in subdirectories
However, once you set directories in the script, you cannot go up those directories
